content/docs/extensions/timescaledb.md

```diff
@@ -0,0 +1,423 @@
+
+
+You can enable the extension by running the following `CREATE EXTENSION` statement in the Neon **SQL Editor** or from a client such as `psql` that is connected to Neon.
+
+
+```sql
+CREATE EXTENSION IF NOT EXISTS timescaledb;
+```
+
+
+For information about using the Neon SQL Editor, see [Query with Neon's SQL Editor](/docs/get-started-with-neon/query-with-neon-sql-editor). For information about using the `psql` client with Neon, see [Connect with psql](/docs/connect/query-with-psql-editor).
+
+
+## Create a Hypertable
+
+
+`timescaledb` hypertables are a high-level abstraction, extending traditional PostgreSQL tables to manage temporal data more effectively. A hypertable simplifies the organization and retrieval of time-series information by providing built-in partitioning based on time intervals.
+
+To begin with, create a SQL table for temperature data:
+
+
+```sql
+CREATE TABLE weather_conditions (
+ time        TIMESTAMP WITH TIME ZONE NOT NULL,
+ device_id   TEXT,
+ temperature  NUMERIC,
+ humidity     NUMERIC
+);
+```
+
+Convert it into a hypertable using the [`create_hypertable`](https://docs.timescale.com/api/latest/hypertable/create_hypertable/) function:
+
+
+```sql
+SELECT create_hypertable('weather_conditions', 'time');
+```
+
+
+You should receive the following output:
+
+
+|       create_hypertable       |
+|-------------------------------|
+| (3,public,weather_conditions,t) |
+
+
+
+It is possible to use both standard SQL commands, and timescaledb functions (which will be covered later).
+
+
+Use a SQL query to insert data in the `weather_conditions` table:
+
+
+```sql
+INSERT INTO weather_conditions
+VALUES
+   (NOW(), 'weather-pro-000002', 72.0, 52.0),
+   (NOW(), 'weather-pro-000003', 71.5, 51.5),
+   (NOW(), 'weather-pro-000004', 73.0, 53.2);
+```
+
+
+Retrieve the data by time in descending order:
+
+
+```sql
+SELECT * FROM weather_conditions ORDER BY time DESC;
+```
+
+You should receive the following output:
+
+
+|             time              |     device_id      | temperature | humidity |
+|-------------------------------|--------------------|-------------|----------|
+| 2024-01-15 13:30:27.464107+00 | weather-pro-000002 |      72.0   |   52.0   |
+| 2024-01-15 13:30:27.464107+00 | weather-pro-000003 |      71.5   |   51.5   |
+| 2024-01-15 13:30:27.464107+00 | weather-pro-000004 |      73.0   |   53.2   |
+
+
+
+
+## Load weather data
+
+Use the [sample weather dataset from TimescaleDB](https://assets.timescale.com/docs/downloads/weather_small.tar.gz) and load it into your Neon database [using psql](/docs/connect/query-with-psql-editor).
+
+
+If you are logged in, quit the `psql` shell:
+
+
+```shell
+\q
+```
+
+
+Download the weather data:
+
+
+```shell
+curl https://assets.timescale.com/docs/downloads/weather_small.tar.gz -o weather_small.tar.gz
+
+
+tar -xvzf weather_small.tar.gz
+```
+
+
+Load the data into Neon database - enter the username, password, host and database name:
+
+
```

Comment:
You can use "shouldwrap" codeblock tags for long commands to avoid the command scrolling off the page:
See https://neon.tech/docs/community/contribution-guide#code-blocks